cmd/root/auth.go

```diff
@@ -67,6 +84,27 @@ func accountClientOrPrompt(ctx context.Context, cfg *config.Config, allowPrompt
 	return a, err
 }
 
+func MustAnyClient(cmd *cobra.Command, args []string) (bool, error) {
+	// Try to create a workspace client
+	werr := MustWorkspaceClient(cmd, args)
+	if werr == nil {
+		return false, nil
+	}
+
+	// If the error is not a workspace client error, return it because configuration is for workspace client
```

Comment:
The comment sounds out of sync with the implementation; if the error is a not a workspace client, we need to fall back to trying to get an account client.

Reply:
@pietern the comment is actually correct because if this is a config for account auth this method will return `databricks.ErrNotWorkspaceClient`. Otherwise, it means it's a workspace config and there is no sense to try it for account

Reply:
Is there a double negation missing needed then?

I.e. if this is not not a workspace client error, and not a no workspace profiles error, then we know the intent is _not_ to create an account client, and can return the error.

Reply:
fixed the comment